### PR TITLE
[Merged by Bors] - feat: port Combinatorics.Additive.Energy

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/Additive/Energy.lean [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+@[to_additive additiveEnergy
+      "The additive energy of two finsets `s` and `t` in a group is the
+      number of quadruples `(a₁, a₂, b₁, b₂) ∈ s × s × t × t` such that `a₁ + b₁ = a₂ + b₂`."]
+def multiplicativeEnergy (s t : Finset α) : ℕ :=
+  (((s ×ᶠ s) ×ᶠ t ×ᶠ t).filter fun x : (α × α) × α × α => x.1.1 * x.2.1 = x.1.2 * x.2.2).card
+#align finset.multiplicative_energy Finset.multiplicativeEnergy
+#align finset.additive_energy Finset.additiveEnergy
+
+@[to_additive additive_energy_mono]
+theorem multiplicative_energy_mono (hs : s₁ ⊆ s₂) (ht : t₁ ⊆ t₂) :
```
Comment (jcommelin) [inline Mathlib/Combinatorics/Additive/Energy.lean:53]: Please also fix the other names
```suggestion
@[to_additive additiveEnergy_mono]
theorem multiplicativeEnergy_mono (hs : s₁ ⊆ s₂) (ht : t₁ ⊆ t₂) :
```

Comment (mo271) [reply Mathlib/Combinatorics/Additive/Energy.lean]: all done

**Comment by mo271** [comment]
Please check that these replacements are ok:
https://github.com/leanprover-community/mathlib4/pull/1702/commits/e777c6d011f104b460c2c93d771f05673d17076c

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**